src/base/searchengine.h

```diff
@@ -38,10 +40,119 @@
 class QProcess;
 class QTimer;
 
+template <class T, std::size_t N>
+class Version
```

Comment:
Because a template allows describing the conception cleanly. How do you suggest to implement conversion to a string? Omit zeroed components too?